include/bluetooth/services/bas_c.h

```diff
@@ -0,0 +1,200 @@
+				      u8_t battery_level,
+				      int err);
+
+/**
+ * @brief Battery Service Client instance.
+ */
+struct bt_gatt_bas_c {
+	/** Connection handle. */
+	struct bt_conn *conn;
+	/** Notification parameters. */
+	struct bt_gatt_subscribe_params notify_params;
+	/** Read parameters. */
+	struct bt_gatt_read_params read_params;
+	/** Notification callback. */
+	bt_gatt_bas_c_notify_cb notify_cb;
+	/** Read value callback. */
+	bt_gatt_bas_c_read_cb read_cb;
+	/** Handle of the Battery Level Characteristic. */
+	u16_t val_handle;
+	/** Handle of the CCCD of the Battery Level Characteristic. */
+	u16_t ccc_handle;
+	/** Current battery value. */
+	u8_t battery_level;
+	/** Properties of the service. */
+	u8_t properties;
+};
+
+/**
+ * @brief Initialize the BAS Client instance.
+ *
+ * You must call this function on the BAS Client object before
+ * any other function.
+ *
+ * @param bas_c  BAS Client object.
+ */
+void bt_gatt_bas_c_init(struct bt_gatt_bas_c *bas_c);
+
+/**
+ * @brief Assign handles to the BAS Client instance.
+ *
+ * This function should be called when a link with a peer has been established,
+ * to associate the link to this instance of the module. This makes it
+ * possible to handle several links and associate each link to a particular
+ * instance of this module. The GATT attribute handles are provided by the
+ * GATT Discovery Manager.
+ *
+ * @note
+ * This function starts the report discovery process.
+ * Wait for one of the functions in @ref bt_gatt_hids_c_init_params
+ * before using the BAS Client object.
+ *
+ * @param dm    Discovery object.
+ * @param bas_c BAS Client object.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ * @retval (-ENOTSUP) Special error code used when the UUID
+ *         of the service does not match the expected UUID.
+ */
+int bt_gatt_bas_c_handles_assign(struct bt_gatt_dm *dm,
+				 struct bt_gatt_bas_c *bas_c);
+
+/**
+ * @brief Subscribe to the battery level value change notification.
+ *
+ * @param bas_c BAS Client object.
+ * @param func  Callback function handler.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int bt_gatt_bas_c_subscribe(struct bt_gatt_bas_c *bas_c,
+			    bt_gatt_bas_c_notify_cb func);
+
+/**
+ * @brief Remove the subscription.
+ *
+ * @param bas_c BAS Client object.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int bt_gatt_bas_c_unsubscribe(struct bt_gatt_bas_c *bas_c);
+
+/**
+ * @brief Get the connection object that is used with a given BAS Client.
+ *
+ * @param bas_c BAS Client object.
+ *
+ * @return Connection object.
+ */
+struct bt_conn *bt_gatt_bas_c_conn(const struct bt_gatt_bas_c *bas_c);
+
+
+/**
+ * @brief Read the battery level value from the device.
+ *
+ * @note The value is stored in the BAS Client object. The notification
```

Comment:
I don't understand what we are trying to say here. The BAS Client object stores the value that is read from the device, the value received from the notification, and the last known value? In what fields?

Reply:
See the comment below.